rcl_logging_spdlog/src/rcl_logging_spdlog.cpp

```diff
@@ -163,19 +164,19 @@ rcl_logging_ret_t rcl_logging_external_initialize(
     }
     int64_t ms_since_epoch = RCUTILS_NS_TO_MS(now);
 
-    // Get the program name.
-    char * basec = rcutils_get_executable_name(allocator);
-    if (basec == nullptr) {
-      // We couldn't get the program name, so get out of here without setting up
-      // logging.
-      RCUTILS_SET_ERROR_MSG("Failed to get the executable name");
-      return RCL_LOGGING_RET_ERROR;
+    bool file_name_provided = (nullptr != file_name_prefix) && (file_name_prefix[0] != '\0');
+    char * basec;
+    if (file_name_provided) {
+      basec = const_cast<char *>(file_name_prefix);
+    } else {  // otherwise, get the program name.
+      basec = rcutils_get_executable_name(allocator);
```

Comment:
falls back to current behavior, so everything should work as before. but this is API break change, needs to rebuild.